llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -1488,26 +1474,30 @@ Error MCDataFragmentMerger::emitMergedFragments() {
   // Use normal node to store the node.
   if (MergeCandidates.size() == 1) {
     auto E = Builder.buildFragment(*MergeCandidates.front().first,
-                                   MergeCandidates.front().second);
+                                   MergeCandidates.front().second,
+                                   MergeCandidatesContents.back());
     reset();
     return E;
   }
   SmallString<8> FragmentData;
   raw_svector_ostream FragmentOS(FragmentData);
-  for (auto &F : MergeCandidates) {
-    switch (F.first->getKind()) {
+  for (uint64_t I = 0; I < MergeCandidates.size(); I++) {
```

Comment:
You could be fancy and write this as
```
for (auto Candidate : llvm:zip(MergeCandidates, MergeCandidatesContents))
```

Reply:
We probably want `(const auto &[Candidate, CandidateContents] : llvm:zip(MergeCandidates, MergeCandidatesContents))`

Reply:
Awesome thanks!